crates/spirv-std/src/textures.rs

```diff
@@ -64,10 +64,9 @@ pub struct StorageImage2d {
 impl StorageImage2d {
     /// Read a texel from an image without a sampler.
     #[spirv_std_macros::gpu_only]
-    pub fn read<I, V, V2, const N: usize>(&self, coordinate: V) -> V2
+    pub fn read<V, V2, const N: usize>(&self, coordinate: V) -> V2
     where
-        I: Integer,
-        V: Vector<I, N>,
+        V: Vector<f32, 2>,
```

Comment:
`coordinate` isn't a `f32` vector, it is an integer vector. Sorry for my typo saying `Vec2` earlier, I meant to point out that it's a 2-component vector and wrote `Vec2` too quickly, I should have written `IVec2/UVec2`.

> Coordinate is an integer scalar or vector containing [...]

Might be best to add tests for this to make sure it compiles correctly.

Reply:
Oh nope yeah that's my bad, I got confused reading the spec, and misread the result type restriction as the coordinate restriction, reverting.